File: 5- Hash Map - Set/C#/1- Find the Difference of Two Arrays.cs
public class Solution {
    public IList<IList<int>> FindDifference(int[] nums1, int[] nums2) {
        HashSet<int> st1 = new HashSet<int>(nums1);
        HashSet<int> st2 = new HashSet<int>(nums2);

        List<int> lst1 = new List<int>() ; 
        foreach(int x in st1){
            if (!st2.Contains(x))
                lst1.Add(x) ;
        }

        List<int> lst2 = new List<int>() ; 
        foreach(int x in st2){
            if (!st1.Contains(x))
                lst2.Add(x) ;
        }
        
        return new List<IList<int>> { lst1, lst2 };
    }
}

File: 05- Hash Map - Set/1- Find the Difference of Two Arrays.cpp
class Solution {
public:
    vector<vector<int>> findDifference(vector<int>& nums1, vector<int>& nums2) {
        unordered_set<int> st1(nums1.begin(), nums1.end());
        unordered_set<int> st2(nums2.begin(), nums2.end());
        
        vector<int> lst1;
        for (int x : st1) {
            if (st2.find(x) == st2.end()) {
                lst1.push_back(x);
            }
        }
        
        vector<int> lst2;
        for (int x : st2) {
            if (st1.find(x) == st1.end()) {
                lst2.push_back(x);
            }
        }
        
        return {lst1, lst2};
    }
};